### PR TITLE
LRU semantic for workdir cache

### PR DESCRIPTION
**Pull request by pfeuffer**
## Proposed changes

Introduces a maximum size for the simple workdir cache. On cache overflow workdirs are evicted using an LRU strategy.
Furthermore parallel requests for the same repository will now block until the workdir is released.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] New code is covered with unit tests
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [x] Feature has been tested with different permissions
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-core/src/main/java/sonia/scm/repository/work/SimpleCachingWorkingCopyPool.java [hunk 1]
```diff
@@ -65,49 +83,142 @@
...
-  private static final Logger LOG = LoggerFactory.getLogger(SimpleCachingWorkingCopyPool.class);
+  public static final int DEFAULT_WORKING_COPY_POOL_SIZE = 5;
+  public static final String WORKING_COPY_POOL_SIZE_PROPERTY = "scm.workingCopyPoolSize";
 
-  private final Map<String, File> workdirs = new ConcurrentHashMap<>();
+  private static final Logger LOG = LoggerFactory.getLogger(SimpleCachingWorkingCopyPool.class);
 
   private final WorkdirProvider workdirProvider;
+  private final LinkedHashMap<String, File> workdirs;
+  private final Map<String, Lock> x;
```
Comment (sdorra) [inline scm-core/src/main/java/sonia/scm/repository/work/SimpleCachingWorkingCopyPool.java:93]: `x`? Why `x`?

File: docs/en/administration/workdir_caching.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (sdorra) [inline docs/en/administration/workdir_caching.md:1]: Don't we need the documentation in German too?

**Review by sdorra — CHANGES_REQUESTED** [review]
When the server is restarted, new working copies are created again and the old ones are neither deleted nor used.

File: docs/en/administration/workdir_caching.md (hunk 2 above)
Comment (pfeuffer) [reply docs/en/administration/workdir_caching.md:1]: I thought this is only mandatory for user documentation (in path `docs/en/user`).?.

File: scm-core/src/main/java/sonia/scm/repository/work/SimpleCachingWorkingCopyPool.java (hunk 1 above)
Comment (pfeuffer) [reply scm-core/src/main/java/sonia/scm/repository/work/SimpleCachingWorkingCopyPool.java]: Ups... Thought too long about a good name.

**Pull request merged**